<?xml version="1.0"?>
<launch>
  <!-- 加载PCD文件的参数 -->
  <arg name="use_pcd_map" default="true"/>
  <arg name="pcd_map_path" default="/root/shared_dir/LIO-SAM/shangpo2map0.01/GlobalMap.pcd"/>
  
  <!-- 坡度代价地图层参数 -->
  <arg name="max_slope_angle" default="30.0"/>
  <arg name="max_roughness" default="0.1"/>
  <arg name="slope_threshold" default="25.0"/>
  <arg name="roughness_threshold" default="0.05"/>
  <arg name="slope_weight" default="0.7"/>
  <arg name="roughness_weight" default="0.3"/>
  <arg name="resolution" default="0.05"/>
  <arg name="visualize" default="true"/>
  
  <!-- 发布静态TF变换 -->
  <node pkg="tf" type="static_transform_publisher" name="map_to_base_link" 
        args="0 0 0 0 0 0 map base_link 100" />
  
  <!-- 坡度代价地图层节点 -->
  <node pkg="costmap_2d" type="costmap_2d_node" name="slope_costmap_node" output="screen">
    <rosparam file="$(find slope_costmap_layer)/config/slope_costmap_params.yaml" command="load" />
    
    <!-- 传递参数到节点 -->
    <param name="use_pcd_map" value="$(arg use_pcd_map)"/>
    <param name="pcd_map_path" value="$(arg pcd_map_path)"/>
    <param name="max_slope_angle" value="$(arg max_slope_angle)"/>
    <param name="max_roughness" value="$(arg max_roughness)"/>
    <param name="slope_threshold" value="$(arg slope_threshold)"/>
    <param name="roughness_threshold" value="$(arg roughness_threshold)"/>
    <param name="slope_weight" value="$(arg slope_weight)"/>
    <param name="roughness_weight" value="$(arg roughness_weight)"/>
    <param name="resolution" value="$(arg resolution)"/>
    <param name="visualize" value="$(arg visualize)"/>
  </node>
  
  <!-- 启动RViz -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find slope_costmap_layer)/rviz/slope_color_visualization.rviz"/>
</launch> 